assets/js/show-trip-details.js

```diff
@@ -1,13 +1,107 @@
 import { updateUsername } from "./user.js";
+import { Place, View } from "./plan.js";
 
 const cardBodies = document.querySelectorAll("div.card div.card-body");
 const myCarousel = document.querySelector('#carouselExampleIndicators');
-
+const saveIcon = document.querySelector("i.bi-bookmark"); 
+const saveTooltip = new bootstrap.Tooltip(saveIcon);
+const profile = document.getElementById("profile");
 
 myCarousel.addEventListener('slide.bs.carousel', function (evt) {
     // show new card body and hide the old one
     cardBodies[evt.from].classList.add("d-none");
     cardBodies[evt.to].classList.remove("d-none");
 })
 
-updateUsername(); 
\ No newline at end of file
+const username = updateUsername(); 
+
+// save bookmark tooltip
+saveTooltip.disable();
+saveIcon.addEventListener('click', postPlanForUser)
+
+async function postPlanForUser() {
+    const url = `/v1/users/${username}/plans`
+    const plan = await getPlan();
+
+    try {
+        console.log("Sending post request to", url)
+        await fetch(
+            url, {
+            method: "POST",
+            headers: {
+                'Content-Type': 'application/json'
+            },
+            body: JSON.stringify(planToView(plan))
+            }
+        )
+    } catch (e) {
+        console.log("Caught an error when saving plan!")
+        console.log("error is:", e) 
```

Comment:
Nit: combine the error logging to one line

Reply:
Done